AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/editor_physx_collider.py

```diff
@@ -0,0 +1,499 @@
+        """
+        _validate_xyz_is_float(x, y, z,
+                               f"One of the Physx Mesh Asset Scale inputs are not a float - X: {x}, Y: {y}, Z {z}")
+        _validate_property_visibility(self.component, self.Path.PhysicsAsset.ASSET_SCALE, PropertyVisibility.VISIBLE)
+
+        self.component.set_component_property_value(self.Path.PhysicsAsset.ASSET_SCALE,
+                                                    math.Vector3(float(x), float(y), float(z)))
+
+    def get_physx_mesh_asset_scale(self) -> None:
+        """
+        Property Type, Default Visibility - ('Vector3', 'Visible')
+
+        Used to get the PhysX Collider's PhysX Mesh Asset Scale.
+        """
+        return self.component.get_component_property_value(self.Path.PhysicsAsset.ASSET_SCALE)
+
+    def set_use_physics_materials_from_asset(self, value: bool) -> None:
+        """
+        Property Type, Default Visibility - ('bool', 'Visible')
+
+        Used to set the PhysX Collider's Physics Materials From Asset. This allows the user to toggle between
+            using the asset provided PhysX Materials, or to override them using the PhysX Collider's PhysX Materials
+            property.
+        """
+        _validate_property_visibility(self.component, self.Path.PhysicsAsset.PHYSICS_MATERIALS_FROM_ASSET,
+                                      PropertyVisibility.VISIBLE)
+
+        self.component.set_component_property_value(self.Path.PhysicsAsset.PHYSICS_MATERIALS_FROM_ASSET, value)
+
+    def get_use_physics_materials_from_asset(self) -> bool:
+        """
+        Property Type, Default Visibility - ('bool', 'Visible')
+
+        Used to get the PhysX Collider's Draw Collider Property value.
+        """
+        return self.component.get_component_property_value(self.Path.PhysicsAsset.PHYSICS_MATERIALS_FROM_ASSET)
+
+    # Shape: Sphere
+    def set_sphere_shape(self) -> None:
+        """
+        Property Type, Default Visibility - ('SphereShapeConfiguration', 'NotVisible')
+
+        Used to set the PhysX Collider's Shape property to Sphere.
+        """
+        self.component.set_component_property_value(self.Path.SHAPE, physics.ShapeType_Sphere)
+
+    def set_sphere_radius(self, radius: float) -> None:
+        """
+        Property Type, Default Visibility - ('float', 'Visible')
+
+        Used to set the PhysX Collider's Sphere radius property.
+        """
+        assert isinstance(radius, float), f"The value passed to Sphere Radius \"{radius}\" is not a float."
+        _validate_property_visibility(self.component, self.Path.Sphere.RADIUS, PropertyVisibility.VISIBLE)
+
+        self.component.set_component_property_value(self.Path.Sphere.RADIUS, radius)
+
+    def get_sphere_radius(self) -> float:
+        """
+        Property Type, Default Visibility - ('float', 'Visible')
+
+        Used to get the PhysX Collider's Sphere radius property.
+        """
+        return self.component.get_component_property_value(self.Path.Sphere.RADIUS)
```

Comment:
question: When the collider is set to another shape type will this return 0? Or will it return whatever cached value happens to have been set last? Just curious as this could be a weird edge case - we might also in future want to consider this returning some kind of optional value that might be 'empty' if the collider is not in the 'sphere' configuration for example.

Reply:
This would return the default values set to these components. I've thought about adding the visibility guard here as well.

Reply:
I see understood, thanks for clarifying, that's not a bad default for sure